src/deroff.rs

```diff
@@ -1,8 +1,40 @@
 /// A translation of https://github.com/fish-shell/fish-shell/blob/e7bfd1d71ca54df726a4f1ea14bd6b0957b75752/share/tools/deroff.py
 // """ Deroff.py, ported to Python from the venerable deroff.c """
+use std::collections::HashMap;
+
+type TODO_TYPE = u8;
+type TODO_NUMBER_TYPE = i8;
+
+const OPTIONS: u8 = 0;
+const FORMAT: u8 = 1;
+const DATA: u8 = 2;
 
 // class Deroffer:
-struct Deroffer {}
+struct Deroffer {
+    reg_table: HashMap<TODO_TYPE, TODO_TYPE>,
+    tr_from: String,
+    tr_to: String,
+    tr: String,
+    nls: TODO_NUMBER_TYPE,
+    specletter: bool,
+    refer: bool,
+    r#macro: TODO_NUMBER_TYPE,
+    nobody: bool,
+    inlist: bool,
+    inheader: bool,
+    pic: bool,
+    tbl: bool,
+    tblstate: TODO_NUMBER_TYPE,
+    tblTab: String,
+    eqn: bool,
+    skipheaders: bool,
+    skiplists: bool,
```

Comment:
Looks like `skiplists` is always `False` in the original code. Can make it a global constant? (Maybe it's a useful toggle somehow in debugging?)